src/csrc/include/internal/rl/off_policy/sac.h

```diff
@@ -114,18 +119,26 @@ void train_sac(const PolicyPack& p_model, const std::vector<ModelPack>& q_models
     alpha_loss.backward();
 
     // reduce gradients
-    if (p_model.comm) {
-      std::vector<torch::Tensor> grads;
-      grads.reserve(alpha_model->parameters().size());
-      for (const auto& p : alpha_model->parameters()) {
-        grads.push_back(p.grad());
+    if ((state->step_train_current + 1) % p_model.grad_accumulation_steps == 0) {
+      if (p_model.comm) {
+        std::vector<torch::Tensor> grads;
+        grads.reserve(alpha_model->parameters().size());
+        for (const auto& p : alpha_model->parameters()) {
+          grads.push_back(p.grad());
+        }
+        p_model.comm->allreduce(grads, true);
       }
-      p_model.comm->allreduce(grads, true);
-    }
 
-    alpha_optimizer->step();
-    if (alpha_lr_scheduler) {
-      alpha_lr_scheduler->step();
+      // I think we do not need grad clipping here
+      // the model is just a scalar and clipping the grad for that might hurt
```

Comment:
Do you know what stablebaselines does for this (if they support grad clipping)? Might be worth matching whatever that library does.

Reply:
They do not support grad clipping nor gradient accumulation at all. So hard to say what they could do.